ecl/eclscheduler/eclscheduler.cpp

```diff
@@ -172,14 +154,200 @@ void openLogFile()
+
+#ifdef _DEBUG
+        return false; // we want full leak checking info
+#else
+        return true; // we don't care - just exit as fast as we can
+#endif
+    }
+
+    void init()
+    {
+        running = true;
+        LocalIAbortHandler abortHandler(*this);
+
+        auto updateFunc = [this](const IPropertyTree *oldComponentConfiguration, const IPropertyTree *oldGlobalConfiguration)
+        {
+            if (running)
+                if (configQueueNamesHasChanged(getComponentConfigSP()))
+                    configChangeOrAbort.signal();
+        };
+        updateConfigCBId = installConfigUpdateHook(updateFunc, true);
+    }
+
+    void run()
+    {
+        while (running)
+        {
+            // wait for either a config change or an abort
+            configChangeOrAbort.wait();
+
+            if (running)
+            {
+                // Process a config change if one is pending
+                try
+                {
+                    CriticalBlock b(updateSchedulersCS);
+                    // onAbort could have been triggered, so check running before any updates are made
+                    if (!running)
+                        break;
+                    updateSchedulerQueues();
+                }
+                catch (IException *E)
+                {
+                    EXCLOG(E);
+                }
+                catch (...)
+                {
+                    IERRLOG("Unknown exception caught in eclscheduler processQueues - restarting");
+                }
+            }
+        }
+        DBGLOG("eclscheduler closing");
+        stop();
+    }
+
+private:
+    bool configQueueNamesHasChanged(const IPropertyTree *config)
+    {
+        std::set<std::string> newSchedulerQueues;
+#ifdef _CONTAINERIZED
+        Owned<IPTreeIterator> queues = config->getElements("queues");
+        ForEach(*queues)
+        {
+            IPTree &queue = queues->query();
+            const char *qname = queue.queryProp("@name");
+            newSchedulerQueues.emplace(qname);
+        }
+#else
+        const char *processName = config->queryProp("@name");
+        Owned<IStringIterator> targetClusters = getTargetClusters("EclSchedulerProcess", processName);
+        ForEach(*targetClusters)
+        {
+            SCMStringBuffer targetCluster;
+            targetClusters->str(targetCluster);
+            newSchedulerQueues.emplace(targetCluster.str());
+        }
+#endif
+        if (newSchedulerQueues.empty())
+            ERRLOG("No queues found to listen on");
+
+        CriticalBlock b(updateSchedulersCS);
+        if (newSchedulerQueues == schedulerQueues)
+        {
+            DBGLOG("Queue names have not changed, not updating");
+            return false;
+        }
+
+        StringBuffer oldNames, newNames;
+        toString(schedulerQueues, oldNames);
+        toString(newSchedulerQueues, newNames);
+        PROGLOG("Updating queue due to queue names change from '%s' to '%s'", oldNames.str(), newNames.str());
```

Comment:
not sure this is (and toString) is worth it, updateSchedulerQueues will also list what is removed and new ones added.